File: 프로그래머스/lv0/120871. 저주의 숫자 3/저주의 숫자 3.js
function solution(n) {
  let i = 1;
  let answer = 1;
  while (true) {
    if (i >= n) {
      break
    }

    i += 1;
    answer += 1;

    while (true) {
      if (answer % 3 === 0) {
        answer += 1;
      } else if (String(answer).includes('3')) {
        answer += 1;
      } else {
        break
      }
    }
  }
  return answer;
}

File: 프로그래머스/lv0/120871. 저주의 숫자 3/저주의 숫자 3.py
def solution(n):
    answer = 1
    count = 1
    
    while count < n:
        count += 1
        answer += 1
        
        # 3의 배수인 경우
        while answer % 3 == 0:
            answer += 1
        # 숫자 3을 사용하는 경우
        while "3" in str(answer):
            answer += 1
        # 3의 배수인 경우
        while answer % 3 == 0:
            answer += 1
        
        print(count,answer)
    print(count,answer, 1)
    return answer